src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/helpers/ListingHelper.java

```diff
@@ -679,6 +679,43 @@ public static String getVenueFromDateListedType(DateListedType dateListedType) {
         return " ";
     }
 
+    public static String getVenueCodeFromDateListedType(DateListedType dateListedType) {
+        // EnglandWales
+        if (dateListedType.hasHearingVenue()) {
+            return dateListedType.getHearingVenueDay().getValue().getCode();
+        }
+
+        // Scotland
+        String hearingVenueScotland = dateListedType.getHearingVenueDayScotland();
+        final TribunalOffice tribunalOffice = TribunalOffice.valueOfOfficeName(hearingVenueScotland);
+        switch (tribunalOffice) {
+            case GLASGOW:
+                if (dateListedType.hasHearingGlasgow()) {
+                    return dateListedType.getHearingGlasgow().getSelectedCode();
+                }
+                break;
+            case ABERDEEN:
+                if (dateListedType.hasHearingAberdeen()) {
+                    return dateListedType.getHearingAberdeen().getSelectedCode();
+                }
+                break;
+            case DUNDEE:
+                if (dateListedType.hasHearingDundee()) {
+                    return dateListedType.getHearingDundee().getSelectedCode();
+                }
+                break;
+            case EDINBURGH:
+                if (dateListedType.hasHearingEdinburgh()) {
+                    return dateListedType.getHearingEdinburgh().getSelectedCode();
+                }
+                break;
+            default:
+                break;
+        }
+
+        return " ";
```

Comment:
I would throw an exception here. Otherwise what does it mean to return " "?
IllegalStateException

Reply:
If the venue is not found then it will match the next hearingVenue. If we throw exception here, it will break that loop. Empty string is returned to set IslistingVenueValid to false and go to next iteration